src/analysis/fourier/poisson_summation.lean

```diff
@@ -112,3 +123,140 @@ begin
+  simp_rw continuous_map.restrict_apply_mk,
+  refine λ y, (hd y.1 (by linarith [hx.1, y.2.1])).trans _,
+  rwa [mul_assoc, mul_le_mul_left hc, norm_of_nonneg, norm_of_nonneg],
+  { convert claim x (by linarith only [hx.1]) y.1 y.2.1,
+    { apply abs_of_nonneg, linarith [y.2.1] },
+    { exact abs_of_pos hx'.1 } },
+  all_goals { exact rpow_nonneg_of_nonneg (abs_nonneg _) _ },
+end
+
+lemma is_O_norm_Icc_restrict_at_bot {f : C(ℝ, E)} {b : ℝ} (hb : 0 < b)
+  (hf : is_O at_bot f (λ x : ℝ, |x| ^ (-b))) (R S : ℝ) :
+  is_O at_bot (λ x : ℝ, ‖f.restrict (Icc (x + R) (x + S))‖) (λ x : ℝ, |x| ^ (-b)) :=
+begin
+  have h1 : is_O at_top (f.comp (continuous_map.mk _ continuous_neg)) (λ x : ℝ, |x| ^ (-b)),
+  { convert hf.comp_tendsto tendsto_neg_at_top_at_bot,
+    ext1 x, simp only [function.comp_app, abs_neg] },
+  have h2 := (is_O_norm_Icc_restrict_at_top hb h1 (-S) (-R)).comp_tendsto tendsto_neg_at_bot_at_top,
+  have : ((λ (x : ℝ), |x| ^ -b) ∘ has_neg.neg) = (λ (x : ℝ), |x| ^ -b),
+  { ext1 x, simp only [function.comp_app, abs_neg] },
+  rw this at h2,
+  refine (is_O_of_le _ (λ x, _)).trans h2, -- equality holds, but less work to prove `≤` alone
+  rw [norm_norm, function.comp_app, norm_norm, continuous_map.norm_le _ (norm_nonneg _)],
+  rintro ⟨x, hx⟩,
+  rw [continuous_map.restrict_apply_mk],
+  refine (le_of_eq _).trans (continuous_map.norm_coe_le_norm _ ⟨-x, _⟩),
+  { exact ⟨by linarith [hx.2], by linarith [hx.1]⟩ },
+  { rw [continuous_map.restrict_apply_mk, continuous_map.comp_apply, continuous_map.coe_mk,
+      neg_neg] }
+end
+
+lemma is_O_norm_restrict_cocompact {f : C(ℝ, E)} {b : ℝ} (hb : 0 < b)
+  (hf : is_O (cocompact ℝ) f (λ x : ℝ, |x| ^ (-b))) (K : compacts ℝ) :
+  is_O (cocompact ℝ) (λ x, ‖(f.comp (continuous_map.add_right x)).restrict K‖) (λ x, |x| ^ (-b)) :=
+begin
+  obtain ⟨r, hr⟩ := K.is_compact.bounded.subset_ball 0,
+  rw [closed_ball_eq_Icc, zero_add, zero_sub] at hr,
+  have : ∀ (x : ℝ), ‖(f.comp (continuous_map.add_right x)).restrict K‖ ≤
+    ‖f.restrict (Icc (x - r) (x + r))‖,
+  { intro x,
+    rw continuous_map.norm_le _ (norm_nonneg _),
+    rintro ⟨y, hy⟩,
+    refine (le_of_eq _).trans (continuous_map.norm_coe_le_norm _ ⟨y + x, _⟩),
+    exact ⟨by linarith [(hr hy).1], by linarith [(hr hy).2]⟩,
+    simp_rw [continuous_map.restrict_apply, continuous_map.comp_apply,
+      continuous_map.coe_add_right, subtype.coe_mk] },
+  simp_rw [cocompact_eq, is_O_sup] at hf ⊢,
+  split,
+  { refine (is_O_of_le at_bot _).trans (is_O_norm_Icc_restrict_at_bot hb hf.1 (-r) r),
+    simp_rw norm_norm, exact this },
+  { refine (is_O_of_le at_top _).trans (is_O_norm_Icc_restrict_at_top hb hf.2 (-r) r),
+    simp_rw norm_norm, exact this },
+end
+
+
+/-- **Poisson's summation formula**, assuming that both `f` decays as
+`|x| ^ (-b)` for some `1 < b` and its Fourier transform is summable. -/
+lemma real.tsum_eq_tsum_fourier_integral_of_rpow_decay_of_summable {f : C(ℝ, ℂ)}
```

Comment:
It was necessary for the previous statements to have a bundled continuous function, as you were using the norms of restrictions which only make sense in this bundled context. However, for the current statement and the next one, bundling is not necessary any more, so can you instead (or also?) give a version for `f : ℝ -> ℂ` with an assumption `continuous f`?